### ISSUE TITLE
DL3009 with RUN --mount

### ISSUE THREAD

**Issue by jonathan-conder-sm** [hadolint/hadolint#723]
- [ ] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

For the Dockerfile below (based on [this one](https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/syntax.md#example-cache-apt-packages)), DL3009 should be suppressed.

### Actual behavior

> Dockerfile:7 DL3009 info: Delete the apt-get lists after installing something

### Steps to reproduce the behavior

**Output of `hadolint --version`:**

```bash
$ hadolint --version
Haskell Dockerfile Linter 2.7.0-no-git
```

**Dockerfile**

```Dockerfile
# syntax = docker/dockerfile:1.3
FROM ubuntu:focal-20211006

RUN rm -f /etc/apt/apt.conf.d/docker-clean; \
    echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache

RUN --mount=type=cache,target=/var/cache/apt \
    --mount=type=cache,target=/var/lib/apt \
    apt-get update && \
    apt-get --no-install-recommends install -y gcc=4:9.3.0-1ubuntu2
```
